sched/scheduler/stateful_scheduler.go

```diff
@@ -92,11 +105,11 @@ func NewStatefulScheduler(
 
 	// TODO: we need to allow the scheduler to accept new jobs
 	// while recovering old ones.
-	if recoverJobsOnStartup {
+	if config.RecoverJobsOnStartup {
 		sched.startUp()
 	}
 
-	if !debugMode {
+	if !config.DebugMode {
```

Comment:
Debug mode needs to be implemented for the whole Scheduler and not just per job, because it controls the execution of the Scheduler's inner loop.  The only difference is if this advances automatically or if we do it manually for debugging